src/com/xilinx/rapidwright/rwroute/RWRoute.java

```diff
@@ -849,7 +848,8 @@ private void setRerouteCriticality() {
 			criticalConnections.sort(new Comparator<Connection>() {
 				@Override
 				public int compare(Connection connection1, Connection connection2) {
-					return connection1.getCriticality() < connection1.getCriticality()? 1 : -1;
+					if(connection1.getCriticality() == connection2.getCriticality()) return 0;
+					else return connection1.getCriticality() < connection2.getCriticality()? 1 : -1;
```

Comment:
```suggestion
					return Float.compare(connection1.getCriticality(),connection2.getCriticality());
```
@YunxZhou - would this code be simpler?  Is it important that the return values be `-1` and `1` or does it just need to be negative and positive integers?

Reply:
See https://docs.oracle.com/javase/7/docs/api/java/lang/Float.html#compare(float,%20float) for API details.

Reply:
See https://docs.oracle.com/javase/7/docs/api/java/lang/Float.html#compare(float,%20float) for API details.

Reply:
@clavin-xlnx, thanks for the comment. Simplifying is good. The compare just needs to be negative and positive integers for the sorting. The order needs to be the other way around. I made a commit based on this